dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/organisationunit/DefaultOrganisationUnitService.java

```diff
@@ -478,20 +492,27 @@ public boolean isInUserHierarchy( User user, OrganisationUnit organisationUnit )
     }
 
     @Override
-    @Transactional(readOnly = true)
+    @Transactional( readOnly = true )
     public boolean isInUserHierarchy( String uid, Set<OrganisationUnit> organisationUnits )
     {
         OrganisationUnit organisationUnit = organisationUnitStore.getByUid( uid );
 
-        return organisationUnit != null && organisationUnit.isDescendant(organisationUnits);
+        return organisationUnit != null && organisationUnit.isDescendant( organisationUnits );
+    }
+
+    @Override
+    @Transactional( readOnly = true )
+    public List<String> getUIDsCreatedBefore( Date date )
+    {
+        return organisationUnitStore.getUIDsCreatedBefore( date );

```

Comment:
Instead of spreading `getUIDsCreatedBefore` method all over services, could you instead just add it to the `IdentifiableObjectManager` interface? This way we centralize it. We are trying to shift away from replicating generic methods in all service interfaces.